Termux download retry timeout

TomJo2000 opened the pull request:
This PR should hopefully cut down on the time spent by `termux_download` when retrying an unresponsive URL, by putting a 300 second cumulative limit on retries.
- See: #23841
- This PR partly follows up on: #21251

TomJo2000 commented:
At least what I've observed with the freedesktop site (which Kitty is trying to pull some data from during its build but can't due to the current migration) it takes about 135 seconds for the first attempt to even finish timing out.
Although to be fair, if the site hasn't responded in 135 seconds it's probably dead.

truboxl commented:
Maybe can add `--connect-timeout`
https://curl.se/docs/manpage.html#--connect-timeout

TomJo2000 commented:
> Maybe can add `--connect-timeout` https://curl.se/docs/manpage.html#--connect-timeout

30 seconds should be a reasonable value for that I think.

TomJo2000 commented:
Pushed an updated commit.
I changed `--retry-max-time` to 120, since we seem to be going by multiples of 30 seconds.
And added `--connect-timeout 30`.

I also put comments on each of the args since some of them are relatively uncommon.

TomJo2000 commented:
I'd like to merge this in 4-6 hours if there's no further comments.
